sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/view/ViewBuilder.java

```diff
@@ -52,19 +56,102 @@ public ViewBuilder setAggregatorFactory(AggregatorFactory aggregatorFactory) {
   }
 
   /**
-   * sets {@link LabelsProcessorFactory}.
+   * Specify the attributes processor for this view.
    *
-   * @param labelsProcessorFactory labels processor factory.
+   * <p>Note: This resets all attribute filters, baggage appending and other processing.
+   *
+   * @param processor The pre-processor for measurement attributes.
+   * @return this Builder.
+   */
+  public ViewBuilder setAttributesProcessor(AttributesProcessor processor) {
+    this.processor = processor;
+    return this;
+  }
+
+  /**
+   * Filters measurement attributes using a given filter.
+   *
+   * <p>Note: This runs after all other attribute processing added so far.
+   *
+   * @param keyFilter filter for key names to include.
+   * @return this Builder.
+   */
+  public ViewBuilder addAttributeKeyFilter(Predicate<String> keyFilter) {
+    this.processor = this.processor.then(AttributesProcessor.filterByKeyName(keyFilter));
+    return this;
+  }
+
+  /**
+   * Filters measurement attributes using a given regex.
+   *
+   * <p>Note: This runs after all other attribute processing added so far.
+   *
+   * @param keyPattern the regular expression for selecting attributes by key name.
+   * @return this Builder.
+   */
+  public ViewBuilder addAttributeKeyPattern(Pattern keyPattern) {
```

Comment:
how about "filterAttributesByKeyPattern"?

Reply:
these names kind of imply an "order" to how they're done.  I was on the fence behind forcing an order or just doing them in the order they're called on the builder.   Do you have a preference?

Reply:
I think just doing them in the order called will probably be the easiest to explain and document.